baseplate/context/sqlalchemy.py

```diff
@@ -95,15 +95,15 @@ class SQLAlchemySessionContextFactory(SQLAlchemyEngineContextFactory):
     :param sqlalchemy.engine.Engine engine: A configured SQLAlchemy engine.
 
     """
-    def make_object_for_context(self, name, server_span):
+    def make_object_for_context(self, name, span):
         engine = super(SQLAlchemySessionContextFactory,
-            self).make_object_for_context(name, server_span)
+            self).make_object_for_context(name, span)
         session = Session(bind=engine)
-        server_span.register(SQLAlchemySessionServerSpanObserver(session))
+        span.register(SQLAlchemySessionSpanObserver(session))
```

Comment:
I think making a new session for local spans would be pretty confusing, particularly if they rollback or whatever at the end of the local span which might be pretty unexpected behaviour. It also makes local spans much more heavyweight of an operation if you have sqlalchemy set up. What do you think of this happening only for server spans?